omnisend-for-ninja-forms/includes/Provider/class-omnisendactionsettingsprovider.php

```diff
@@ -0,0 +1,221 @@
+<?php
+/**
+ * Omnisend Action setting provider
+ *
+ * @package OmnisendNinjaFormsPlugin
+ */
+
+declare(strict_types=1);
+
+namespace Omnisend\NinjaFormsAddon\Provider;
+
+use Omnisend\NinjaFormsAddon\Actions\OmnisendAddOnAction;
+
+/**
+ * Class OmnisendActionSettingsProvider
+ */
+class OmnisendActionSettingsProvider {
+
+	public const SEND_WELCOME_EMAIL = 'send_welcome_email_checkbox';
+	private const TYPES             = array(
+		'email'       => OmnisendAddOnAction::EMAIL,
+		'address'     => OmnisendAddOnAction::ADDRESS,
+		'city'        => OmnisendAddOnAction::CITY,
+		'listcountry' => OmnisendAddOnAction::COUNTRY,
+		'firstname'   => OmnisendAddOnAction::FIRST_NAME,
+		'lastname'    => OmnisendAddOnAction::LAST_NAME,
+		'phone'       => OmnisendAddOnAction::PHONE_NUMBER,
+		'date'        => OmnisendAddOnAction::BIRTHDAY,
+		'zip'         => OmnisendAddOnAction::POSTAL_CODE,
+		'checkbox'    => 'checkbox',
+	);
+
+	/**
+	 * Get the settings for the OmnisendActionSettingsProvider.
+	 *
+	 * @return array The settings array.
+	 */
+	public function get_settings(): array {
+		if ( ! isset( $_GET['form_id'] ) || wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['form_id'] ) ) ) ) {
```

Comment:
Not sure, but I am suggested to have the following check:
```
// Assuming a nonce is passed as a separate parameter in the $_GET array.
if ( ! isset( $_GET['form_id'], $_GET['nonce'] ) || ! wp_verify_nonce( $_GET['nonce'], 'your_nonce_action' ) ) {
    return array();
}
```

Reply:
The phpcf added this check there. The problem is that ninja forms doesn't add a nonce to request, so I can't check it. The source code from ninja forms, how they handle form id - https://i.imgur.com/62cOUEa.png